integration/constraint_test.go

```diff
@@ -35,7 +36,16 @@ func (s *ConstraintSuite) SetUpSuite(c *check.C) {
 	s.consulClient = consulClient
 
 	// Wait for consul to elect itself leader
-	time.Sleep(2000 * time.Millisecond)
+	err = try.Do(3*time.Second, func() error {
+		leader, err := consulClient.Status().Leader()
+
+		if err != nil || len(leader) == 0 {
+			return fmt.Errorf("Leader not find. %v", err)
+		}
+
+		return nil
+	})
+	c.Assert(err, checker.IsNil)
```

Comment:
(Note that this is the reason why I think `try.Do(t, 3*time.Second, …)` would fit perfectly. But can be done in follow-ups.